internal/secrets/resolver.go

```diff
@@ -20,68 +19,70 @@ const (
 )
 
 type SecretResolver struct {
-	ManagementNamespace    string
-	ReleaseNamespace       string
-	EnableNamespaceSecrets bool
-	Client                 client.Client
-	Log                    logr.Logger
+	EnableMultipleSubaccounts bool
+	ManagementNamespace       string
+	ReleaseNamespace          string
+	EnableNamespaceSecrets    bool
+	Client                    client.Client
+	Log                       logr.Logger
 }
 
-func (sr *SecretResolver) GetSecretForResource(ctx context.Context, namespace, name string) (*v1.Secret, error) {
-	var secretForResource *v1.Secret
-	var err error
-	found := false
+func (sr *SecretResolver) GetSecretForResource(ctx context.Context, namespace, name, subaccountID string) (*v1.Secret, error) {
+	secretForResource := &v1.Secret{}
 
-	if sr.EnableNamespaceSecrets {
-		sr.Log.Info("Searching for secret in resource namespace", "namespace", namespace, "name", name)
-		secretForResource, err = sr.getSecretFromNamespace(ctx, namespace, name)
-		if client.IgnoreNotFound(err) != nil {
-			sr.Log.Error(err, "Could not fetch secret in resource namespace")
-			return nil, err
-		}
-
-		found = !apierrors.IsNotFound(err)
+	if !sr.EnableMultipleSubaccounts {
+		return sr.getDefaultSecret(ctx, name)
```

Comment:
add log.info that EnableMultipleSubaccounts is disabled and therefore using default

Reply:
done